5-parameterization/README.md

```diff
@@ -5,116 +5,127 @@ The API becomes more powerful in this lab, but we want to be careful and only ro
+```
+
+We extended the _API_ module by introducing a new input variable `enable_greeting_feature`. The default is set to `false`, so we can’t accidentally distribute the new feature. In the `main.tf` file, we simply pass the input variable down to the AWS Lambda function as an environment variable. Finally, in the Lambda function, we use the environment variable to flip on the new feature.
 
 The new feature wouldn’t appear after deployment (feel free to try it and deploy your staging and production environment). We need to configure the new input variable explicitly. Let’s do it.
 
 ## Rollout
 
 1. Go to the file `staging/main.tf` and replace it:
-  ```tf
-  terraform {
-    required_version = "~> 1.1.7"
 
-    backend "s3" {
-      key    = "staging/terraform.tfstate"
-      region = "eu-west-1"
-    }
-  }
+```tf
+terraform {
+  required_version = "~> 1.1.7"
 
-  provider "aws" {
+  backend "s3" {
+    key    = "staging/terraform.tfstate"
     region = "eu-west-1"
   }
+}
 
-  module "website" {
-    source = "../modules/website"
+provider "aws" {
+  region = "eu-west-1"
+}
 
-    environment = "staging"
-  }
+module "website" {
+  source = "../modules/website"
 
-  module "api" {
-    source = "../modules/api"
+  environment = "staging"
+}
+
+module "api" {
+  source = "../modules/api"
+
+  environment             = "staging"
+  enable_greeting_feature = true
+}
+```
 
-    environment             = "staging"
-    enable_greeting_feature = true
-  }
-  ```
 2. Cd into the `staging` folder and run `terraform apply`. Confirm with `yes`.
 3. Open the API URL in the browser. You should still see this message:
-  ```json
-  {"message":"Hello from Lambda! 👋"}
-  ```
+
+```json
+{ "message": "Hello from Lambda! 👋" }
+```
+
 4. Now, add the `name` query param to the URL, e.g.:
-  ```
-  https://XXXXXXXXXX.execute-api.eu-west-1.amazonaws.com/?name=alice
-  ```
+
+```
+https://XXXXXXXXXX.execute-api.eu-west-1.amazonaws.com/?name=alice
+```
+
 5. Here we go! The new feature works on staging.
 
 With input variables, we can make modules configurable for different scenarios. In this case, we only want to deploy a new feature to the staging environment, but not to production. In practice, it’s a common requirement to configure environments differently. For example, we want to configure provisioned capacities (like CPU or memory allocation), a global CDN or custom domains with SSL certificates.
 
 ## Final words
 
-Well, that was the last lab for the Terraform workshop. We hope you enjoyed the workshop. If you want to learn more about Terraform and dive deeper, take a look at the [reading list](../README.md#📖-further-reading).
+Well, that was the last lab for the Terraform workshop. We hope you enjoyed the workshop. Before you go, don't forget to remove the stack deployed in your aws account with the `terraform destroy` command. If you want to learn more about Terraform and dive deeper, take a look at the [reading list](../README.md#📖-further-reading).
```

Comment:
I like that short hint. Maybe we can mention that you need to run the command per environment. Also, deleting the remote backend is not covered at all, so you need to delete the S3 bucket manually.

Reply:
Maybe it's worth having a `Clean up` section before `Final words`?

Reply:
It is probably not a stack (term from Cloudformation), maybe "resources"?

Reply:
I included a new clean up section with more detailed instructions as to how remove resources, including the S3 bucket for the state file.